amethyst_input/src/input_handler.rs

```diff
@@ -389,6 +393,43 @@ impl<T: BindingTypes> InputHandler<T> {
     pub fn send_frame_begin(&mut self) {
         self.mouse_wheel_vertical = 0.0;
         self.mouse_wheel_horizontal = 0.0;
+
+        // Prepare the mouse axis input
+        {
+            let hits_boundaries = |delta: f64| delta >= 1. || delta <= -1.;
+            let normalize = |x: f64| {
+                if x > 1. {
+                    return 1.;
+                } else if x < -1. {
+                    return -1.;
+                }
+                return x;
+            };
+            let current_pos = self.mouse_position.unwrap_or((0., 0.));
+            let last_pos = self.mouse_last_position.unwrap_or((0., 0.));
+            let raw_delta_x = current_pos.0 - last_pos.0;
+            let raw_delta_y = current_pos.1 - last_pos.1;
+            let delta_x = normalize(raw_delta_x);
+            let delta_y = normalize(raw_delta_y);
+
+            // In case the deltas hit the boundaries, not all data could be handled in the last frame,
+            // so the buffer should be updated with the overflow
+            if hits_boundaries(delta_x) {
+                self.mouse_axis_buffer.0 = raw_delta_x - delta_x;
```

Comment:
If we move say, 5 pixels in a frame, wouldn't this code as is only get 2 of those 5 pixels?

Once for the first frame, again for the second frame, and then the buffer gets set to zero.

Reply:
The other problem with this is that even if we correct this code we're reporting several frames of maximum axis values if the user moves, for example,15 pixels in a frame.

So what this would look like is

1. User moves mouse kind of quick
2. at 60 FPS, this axis is maxed out for a full quarter of a second.

Reply:
> If we move say, 5 pixels in a frame, wouldn't this code as is only get 2 of those 5 pixels?

It should not, as we only subtract a maximum of one pixel per frame from the buffer.

> we're reporting several frames of maximum axis values if the user moves

exactly, and I was critical of it, too, however Overmind was of the opinion that it might come in handy - if I understood him correctly. I never really had to use buffered mouse input though...